### ISSUE TITLE
[Task] Default loki-mixin dashboards to TSDB

### ISSUE THREAD

**Issue by trevorwhitney** [grafana/loki#9772]

**Comment by pooladkhay** [comment]
Hi, I'd like to start working on this issue. Can I get a bit of help?

**Comment by trevorwhitney** [comment]
@pooladkhay sure thing!

So I think the best way to accomplish this is by adding a config section for setting the schema or enabling tsdb, [like we have for our scalable deployment (aka ssd)](https://github.com/grafana/loki/blob/f5992274f4c41544cbe6f677dfa1f4d909b12193/production/loki-mixin/config.libsonnet#L24)

Then, in the relevant dashboard libsonnet, we can hide/show rows or panels, [like here](https://github.com/grafana/loki/blob/f5992274f4c41544cbe6f677dfa1f4d909b12193/production/loki-mixin/dashboards/loki-operational.libsonnet#L16)

It may require adding new rows to the raw dashboard json file, and then toggling their visibility in the libsonnet based on the config values set.

Once that is ready, you can change the defaults in [mixin.libsonnet](https://github.com/grafana/loki/blob/f5992274f4c41544cbe6f677dfa1f4d909b12193/production/loki-mixin/mixin.libsonnet#L4) to set the appropriate configs, sort of like what we do in [mixin-ssd.libsonnet](https://github.com/grafana/loki/blob/f5992274f4c41544cbe6f677dfa1f4d909b12193/production/loki-mixin/mixin-ssd.libsonnet#L2).

Hope that helps!

**Comment by trevorwhitney** [comment]
@pooladkhay are you picking this up? Don't want to step on toes if you are, but we have some free cycles over here and were thinking if picking it up if you're not.

**Comment by pooladkhay** [comment]
> @pooladkhay are you picking this up? Don't want to step on toes if you are, but we have some free cycles over here and were thinking if picking it up if you're not.

@trevorwhitney Please go ahead, I'm a bit busy unfortunately

**Comment by hainenber** [comment]
hi @trevorwhitney, has you or your team picking up this yet? If not , I can volunteer to work on. Thanks!

**Comment by hainenber** [comment]
Hi Trevor, it appears that currently there are no metrics to monitor TSDB shipper (see #9719), so my best effort for this task is to add placeholder settings

**Comment by Mahii25** [comment]
is this issue still open? I would love to work on it

**Comment by akuthotaaravind** [comment]
Hey @trevorwhitney i would like to contribute to the issue and work on it .please let me know anyone else working on it or not .

**Comment by trevorwhitney** [comment]
No, I don't think anyone is working on this at the moment. I'm not sure what the current state of the issue is either. https://github.com/grafana/loki/pull/11087 added a `tsdb` boolean in the mixin config, which is used to govern the display of the structured metadata row. Are there other rows we need to put behind the same condition?

**Comment by mag123c** [comment]
I've implemented this feature and submitted PR

## Summary
- Added `tsdb` configuration variable to `config.libsonnet` (defaults to `true`)
- Implemented conditional rendering to show/hide TSDB vs BoltDB dashboards based on configuration

## Context
I noticed this issue has been open for 2 years with multiple attempts. Since TSDB is now the recommended index format, having it as the default in dashboards seems important for new users.

The implementation follows the existing pattern used for SSD mode configuration, making it consistent with the codebase.

Would appreciate a review when you have time. Please let me know if this approach aligns with what you had in mind for this issue, or if you'd prefer a different implementation strategy. Happy to make any adjustments based on your feedback!

cc @trevorwhitney

**Comment by injae-kim** [comment]
Fix PR: https://github.com/grafana/loki/pull/18732 👍
